storage/innobase/ddl/ddl0merge.cc

```diff
@@ -158,6 +158,8 @@ struct Merge_file_sort::Output_file : private ut::Non_copyable {
   /** @return number of rows in the output file. */
   [[nodiscard]] uint64_t get_n_rows() const noexcept { return m_n_rows; }
 
+  Write_offsets m_write_offsets;
```

Comment:
Any chance we can predict the number of potential writes and preallocate `m_write_offsets`
`m_write_offsets.reserve()` (probably in the `init()` method)?

Reply:
There is no easy way. @percona-ysorokin , please, consider that it depends on the size of the new index and of the configuration variables `--innodb-ddl-threads` and `--innodb-ddl-buffer-size`.

 Anyway, the performance improvement of calling `reserve` in advance is insignificant considering all the operations are carried out during the `ALTER TABLE` command.